### PR TITLE
Update hpc slurm gromac example and references to use Slurm V6

### PR DESCRIPTION
**Pull request by harshthakkar01**
### Submission Checklist

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cloud HPC Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: community/examples/hpc-slurm-gromacs.yaml [hunk 1]
```diff
@@ -68,12 +69,6 @@ deployment_groups:
                 projections:
                   all: '{name}/{version}-{compiler.name}-{compiler.version}'
       commands: |
-        # Un-comment and update mirror_url to install from spack cache
-        # if ! spack mirror list | grep -q gcs_cache; then
-        #   spack mirror add --scope site gcs_cache gs://optionally_set_spack_cache_bucket
-        # fi
-        # spack buildcache keys --install --trust
-
```
Comment (nick-stroud) [inline community/examples/hpc-slurm-gromacs.yaml:76]: I would leave this. Was there a reason to remove it?

File: tools/cloud-build/daily-tests/tests/spack-gromacs.yml [hunk 2]
```diff
@@ -15,14 +15,14 @@
 ---
 
 test_name: hpc-slurm-gromacs
-deployment_name: "spack-gromacs-{{ build }}"
+deployment_name: "spackgroma"
```
Comment (nick-stroud) [inline tools/cloud-build/daily-tests/tests/spack-gromacs.yml:18]: If you don't have the build id in here then only 1 test can run in parallel. Please try to include build id in name.

**Review by nick-stroud — CHANGES_REQUESTED** [review]

File: community/examples/hpc-slurm-gromacs.yaml (hunk 1 above)
Comment (harshthakkar01) [reply community/examples/hpc-slurm-gromacs.yaml:76]: Added back.

File: tools/cloud-build/daily-tests/tests/spack-gromacs.yml (hunk 2 above)
Comment (harshthakkar01) [reply tools/cloud-build/daily-tests/tests/spack-gromacs.yml]: Included build name.

**Review by nick-stroud — APPROVED** [review]

File: community/examples/hpc-slurm-gromacs.yaml [hunk 3]
```diff
@@ -68,11 +69,11 @@ deployment_groups:
...
-        # Un-comment and update mirror_url to install from spack cache
-        # if ! spack mirror list | grep -q gcs_cache; then
-        #   spack mirror add --scope site gcs_cache gs://optionally_set_spack_cache_bucket
-        # fi
-        # spack buildcache keys --install --trust
+        ## Un-comment and update mirror_url to install from spack cache
+        ## if ! spack mirror list | grep -q gcs_cache; then
+        ##   spack mirror add --scope site gcs_cache gs://optionally_set_spack_cache_bucket
+        ## fi
+        ## spack buildcache keys --install --trust
```
[the remark this answers is not in the document]
Comment (harshthakkar01) [reply community/examples/hpc-slurm-gromacs.yaml:76]: Had an offline discussion and decided to keep this for now.

**Pull request merged**